### DISCUSSION TITLE
Short Jumps are moving 36 seconds instead of 10

### DISCUSSION THREAD

**Discussion by zitot** [smplayer-dev/smplayer#531] [Q&A]
Just downloaded SMPlayer for Windows, I did install the MPV from the SMPlayer installer as the website said to on https://www.smplayer.info/en/mpv 

I'm playing an .mp4 file. Using Mplayer as my multimedia engine instead of Mpv results in this potential bug? 

Ahh, using Mpv with precise seek fixed it. I don't like how slow precise seek is, but I suppose that's a CPU/RAM/SSD upgrade.

Version: 22.7.0
Operating System: Windows 10

**Comment by smp-marco** [comment]
I've just asked a similar question, but concerning +- 1 second steps https://github.com/smplayer-dev/smplayer/discussions/539
Could you check to see if you are getting this same bug?
I am already using MPV.
Thanks

**Comment by smplayer-dev** [comment]
Try enabling precise seeking in preferences -> interface -> seeking.

**Reply by smp-marco** [reply]
Yes!
It was set to absolute seeking.
I reset it to precise, and it's perfect (not at all slow).

This brings me joy 👍 

Thanks for resolving this for me so rapidly.

**Comment by Peter-202122** [comment]
I can confirm both problems reported by zitot in his post:

a) When "Precise seeking" is not activated, the jumps differ a lot from what is chosen in the preferences (in my case: 5 seconds). And the difference is not the same every time. Sometimes the jump is 8 seconds instead of 5 seconds (which would be acceptable for me) but sometimes is is 20 seconds or even more. Which such a big difference the choice of a certain interval makes no sense any longer. 

And I have the impression that it got worse now after my recent update from version 19.10.0 to version 22.7.0 (using the portable version). (OS is Windows 10, 64 bit).

b) When "Precise seeking" is activated this problem is indeed resolved. But then the delay in performing the jump is by far too big: It takes about 5 to 10 seconds until the desired position is located.

The SMPlayer is really fine. (Thanks for it :-). ) But I need to perform such jumps very often and with these issues it is not really useable for me. Or at least I would have to try to work again with the old version 19.10.0. 

So it would be fine if at least one of the two described issues could be fixed. Many thanks.

PS:
Here a small update:
- In the meantime I reverted to version 19.10.0 which apparently has a better "jumping behaviour" than 22.7.0: In 22.7.0 I have partly the significantly bigger jumps (as described above). 
- On the other hand this seems to refer primarily (or only) to jumps **_backwards_**. Jumps **forward** in most of the cases do not work at all! And as I see now this refers not only to version 22.7.0 but also to version 19.10.0.  [Always speaking about the situation when "Precise seeking" is not activated.]
- I forgot to mention that my version 19.10.0 was (and is again) 32 bit, whereas 22.7.0 was 64 bit. (In case that this could be of importance to explain the different behaviour.) And I always refer to video files (commercial DVDs); not tested with audio files.

**Reply by smp-marco** [reply]
Version: 22.7.0 (revision 10091) (64-bit)
Using Qt 5.15.2 (compiled with Qt 5.15.2)
Using MPV 0.34.0

After 3 weeks of constant usage, my feedback on jumping forward and back is...

**1 second jumps** are precise and immediate (using MP4 format).
They can be multi-clicked without problem.

Note: 
By precise, I mean that 1 second jumps can be made, but I can't confirm frame accuracy.
(I would love to see the position display showing sub-second timing ... that would be so useful)

This changed yesterday, when multi-clicking caused temporary display of Notepad++ , sometimes leaving Notepad++ in the foreground.
It may be that the PC was under stress - I'm not sure, because I was focused on the project.
I closed the day with a 'boot defrag'.

I'll report back tomorrow, on whether this was just a memory/cpu issue.

**Reply by smplayer-dev** [reply]
> I can confirm both problems reported by zitot in his post:
> 
> a) When "Precise seeking" is not activated, the jumps differ a lot from what is chosen in the preferences (in my case: 5 seconds). And the difference is not the same every time. Sometimes the jump is 8 seconds instead of 5 seconds (which would be acceptable for me) but sometimes is is 20 seconds or even more. Which such a big difference the choice of a certain interval makes no sense any longer.
> 
> And I have the impression that it got worse now after my recent update from version 19.10.0 to version 22.7.0 (using the portable version). (OS is Windows 10, 64 bit).
> 
> b) When "Precise seeking" is activated this problem is indeed resolved. But then the delay in performing the jump is by far too big: It takes about 5 to 10 seconds until the desired position is located.
> 
> The SMPlayer is really fine. (Thanks for it :-). ) But I need to perform such jumps very often and with these issues it is not really useable for me. Or at least I would have to try to work again with the old version 19.10.0.
> 
> So it would be fine if at least one of the two described issues could be fixed. Many thanks.
> 
> PS: Here a small update:
> 
> * In the meantime I reverted to version 19.10.0 which apparently has a better "jumping behaviour" than 22.7.0: In 22.7.0 I have partly the significantly bigger jumps (as described above).
> * On the other hand this seems to refer primarily (or only) to jumps **_backwards_**. Jumps **forward** in most of the cases do not work at all! And as I see now this refers not only to version 22.7.0 but also to version 19.10.0.  [Always speaking about the situation when "Precise seeking" is not activated.]
> * I forgot to mention that my version 19.10.0 was (and is again) 32 bit, whereas 22.7.0 was 64 bit. (In case that this could be of importance to explain the different behaviour.) And I always refer to video files (commercial DVDs); not tested with audio files.

Seeking is done by mpv, so I think I can't do much to fix the problem. You can try new or older versions of mpv and see if there's any which works better: https://sourceforge.net/projects/mpv-player-windows/files/release/

**Reply by Peter-202122** [reply]
Thanks for your feedback, smp-marco and smplayer-dev. It's a pity because apart from this inconvenience SMPlayer is very pleasant and comfortable to use. I will follow the advice to try another version of mpv.

**Reply by smp-marco** [reply]
With regard to my previous comment 
" when multi-clicking caused temporary display of Notepad++ , sometimes leaving Notepad++ in the foreground."

I can confirm that this is still occurring.
I recognise that I should accustom myself to using the drop down option, and selecting a longer jump.
However, it may be useful to devs to learn that when jumping back, the focus of the SMPlayer window, apparently switches to other windows (momentarily).

The project has been completed now, so over the next few days I will have a look at this issue, when under less stress :)
